### PR TITLE
Fix OSDK_LOCAL_DEV and let CI test on local versions.

### PR DESCRIPTION
**Pull request by junyang-zh**
Fix the recent CI failure on OSDK test: `test_examples_in_book`.

Originally the CI wants to test the local OSDK with remote published `osdk-test-kernel`, which leads to dual existence of OSTD. Now this PR always uses the local development version of every crate (OSTD, OSDK, `osdk-test-kernel`) for CI.

The updated behavior is desired, we do not actually want to require local modifications of OSDK always compatible with the published `crates.io`'s `osdk-test-kernel` (vice versa). So does CI. CI now uses all local crates. The downside is that we don't really ensure what users get are working, by the current CI infrastructures. `OSDK_LOCAL_DEV`, though used conservatively for testing, would lead to the differences between the local and the published version.

The way to ensure what users get are working seems to add another CI workflow post-publishing (or daily), which may be out of the scope of this PR.

### CODE REVIEW

File: .github/workflows/test_osdk.yml [hunk 1]
```diff
@@ -74,7 +71,4 @@ jobs:
       # So the RUSTUP_HOME needs to be set here. 
       - name: Unit test
         id: unit_test
-        run: |
-          cd osdk
-          RUSTUP_HOME=/root/.rustup INTEL_TDX=1 cargo build
-          RUSTUP_HOME=/root/.rustup INTEL_TDX=1 cargo test
+        run: RUSTUP_HOME=/root/.rustup make test_osdk
```
Comment (tatetian) [inline .github/workflows/test_osdk.yml:74]: You didn't mean to remove  `INTEL_TDX=1`, did you?

**Comment by junyang-zh** [comment]
Update (the 3rd commit): `is_tdx_enabled()` returns `true` when `INTEL_TDX=0`. Makes people wonder why it succeeded earlier.

**Comment by tatetian** [comment]
> Update (the 3rd commit): `is_tdx_enabled()` returns `true` when `INTEL_TDX=0`. Makes people wonder why it succeeded earlier.

When TDX is not enabled, the env var `INTEL_TDX` is not even defined. So the original (incorrect) implementation still works.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**